commerce_coordinator/apps/ecommerce/pipeline.py

```diff
@@ -0,0 +1,21 @@
+"""
+order app filter pipeline implementations
+"""
+
+from openedx_filters import PipelineStep
+
+from commerce_coordinator.apps.ecommerce.clients import EcommerceApiClient
+
+
+class GetEcommerceOrders(PipelineStep):
+    """
+    Adds ecommerce orders to the order data list.
+    """
+    def run_filter(self, order_data, *args, **kwargs):  # pylint: disable=arguments-differ
```

Comment:
thanks, Phil! I was following the example run_filter from demo_lms, where we override the arguments for some additional checking (i.e. make sure order_data is passed in), but let me look closer and see if I can trim this down and make the arguments more similar and less confusing.

Reply:
Update: good question on behavior if any of the params happen to be `None`. Depending on the spot in the code, I'm not sure they're always required. 

These parameters have always been passed long when when frontend-app-ecommerce calls the ecommerce orders endpoint on the way to order history page (username to specify which orders to get from ecommerce, and the other two parameters used for pagination on the order history page), but there are intermediate steps where we appear to grab the authenticated user instead of necessarily needing them set/passed as querystring params. For example: 
- when I replace the ORDER_HISTORY_GET_PARAMETERS (saved [here](https://github.com/edx/commerce-coordinator/blob/d2f96417483eec44b041f6cd63d51ee527537dc3/commerce_coordinator/apps/frontend_app_ecommerce/tests/__init__.py#L4) for tests) with {none} the existing tests in test_views.py still pass(!)
- the commerce coordinator [test page]([url](http://localhost:8140/orders/order_history/?username=edx&page=1&page_size=20)) no longer uses/needs querystring parameters (it shows orders for the logged in user)

So I think in general we need the incoming requests to EcommerceUserOrdersView to have a username (otherwise there's a PermissionDenied error [here](https://github.com/edx/commerce-coordinator/blob/0a252a59604ec7d96a285fc9c3e732dc62964168/commerce_coordinator/apps/orders/views.py#L26)). I think this suggestion is to verify that the parameters make it all the way through to ecommerce, now that there are more steps in the chain. So far the tests have been for individual steps, need to think about an 'end-to-end'-ish test.

Reply:
Very interesting! Thanks for the write up. I definitely can see integration tests being important in the future.